modules/porous_flow/src/fvkernels/FVPorousFlowHeatConduction.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVPorousFlowHeatConduction.h"
+#include "PorousFlowDictator.h"
+
+registerADMooseObject("PorousFlowApp", FVPorousFlowHeatConduction);
+
+InputParameters
+FVPorousFlowHeatConduction::validParams()
+{
+  InputParameters params = FVFluxKernel::validParams();
+  params.addRequiredParam<UserObjectName>("PorousFlowDictator",
+                                          "The PorousFlowDictator UserObject");
+  params.set<unsigned short>("ghost_layers") = 2;
+  params.addClassDescription("Conductive heat flux");
+  return params;
+}
+
+FVPorousFlowHeatConduction::FVPorousFlowHeatConduction(const InputParameters & params)
+  : FVFluxKernel(params),
+    _dictator(getUserObject<PorousFlowDictator>("PorousFlowDictator")),
+    _lambda_element(getADMaterialProperty<RealTensorValue>("PorousFlow_thermal_conductivity_qp")),
+    _lambda_neighbor(
+        getNeighborADMaterialProperty<RealTensorValue>("PorousFlow_thermal_conductivity_qp")),
+    _temperature_element(getADMaterialProperty<Real>("PorousFlow_temperature_qp")),
+    _temperature_neighbor(getNeighborADMaterialProperty<Real>("PorousFlow_temperature_qp")),
+    _grad_T(getADMaterialProperty<RealGradient>("PorousFlow_grad_temperature_qp"))
+{
+}
+
+ADReal
+FVPorousFlowHeatConduction::computeQpResidual()
+{
+  ADRealGradient gradT;
+  ADRealTensorValue coeff;
+
+  // If we are on a boundary face, use the reconstructed gradient computed in _grad_T
+  if (onBoundary(*_face_info))
+  {
+    gradT = -_grad_T[_qp];
```

Comment:
this seems like the element value? not reconstructed

Reply:
This comes from `adGradSln` in a material, which I thought was reconstructed? I'll just simplify this to say `use the gradient computed in _grad_T`

Reply:
I think the grad you are calling here is forwarded through a material property and boils down to this one:
```
   _grad_temperature_var(_nodal_material ? nullptr
                                          : &coupledGenericGradient<is_ad>("temperature")),
```
then tracing that one it's _ad_grad_u which is set by
```
  if (_need_ad_grad_u)
    assignForAllQps(static_cast<const MooseVariableFV<OutputType> &>(_var).adGradSln(
                        _elem, Moose::currentState()),
                    _ad_grad_u,
                    nqp);
```

and the definition of this uses the greenGaussUtility, which does do the reconstruction at the boundary. So yes!